File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.java
class Solution {
    public int[] solution(int num, int total) {
        int[] answer = new int[num];
        int start = 0;

        if (num % 2 == 0){
            start = (total + (num / 2)) / num - num / 2;
        }
        else{
            start = total / num - (num / 2);
        }

        for (int i = 0; i < num; i++) {
            answer[i] = start;
            start++;
        }
        return answer;
    }
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.swift
import Foundation

func solution(_ num:Int, _ total:Int) -> [Int] {
    var answer : [Int] = []
    var start = 0
    
    if (num % 2 == 0){
        start = (total + (num / 2)) / num - num / 2
    }
    else{
        start = total / num - (num / 2)
    }

    for i in 0..<num {
        answer.append(start)
        start += 1
        
    }
    return answer
}